totalRP3/core/impl/communication_protocol_broadcast.lua

```diff
@@ -271,24 +271,65 @@ local function hideBroadcastChannelFromChatFrame()
 	RemoveChatWindowChannel(chatFrame:GetID(), broadcastChannelName);
 end
 
+--- Swap channels by index, without losing the color association on Retail.
+local swapChannelsByIndex = ChatConfigChannelSettings_SwapChannelsByIndex or C_ChatInfo.SwapChatChannelsByChannelIndex;
+
 --- Makes sure the broadcast channel is always at the bottom of list.
 --- This is so the user always have the channels they actually use first and that the broadcast channel
 --- is never taking the General or Trade chat position.
 local function moveBroadcastChannelToTheBottomOfTheList(forceMove)
 	if getConfigValue(TRP3_API.ADVANCED_SETTINGS_KEYS.MAKE_SURE_BROADCAST_CHANNEL_IS_LAST) and (forceMove or helloWorlded) then
-		local broadcastChannelName = config_BroadcastChannel();
-
-		for channelIndex = 1, MAX_WOW_CHAT_CHANNELS do
-			local _, channelName = GetChannelName(channelIndex);
-			if channelName == broadcastChannelName then
-				(SwapChatChannelByLocalID or C_ChatInfo.SwapChatChannelsByChannelIndex)(channelIndex, channelIndex + 1);
+		local broadcastChannelIndex = GetChannelName(config_BroadcastChannel());
+		if broadcastChannelIndex == nil then return end
+
+		-- Get the index of the last channel
+		local lastChannelIndex = 0;
+		for index = broadcastChannelIndex, MAX_WOW_CHAT_CHANNELS do
+			if (C_ChatInfo.GetChannelShortcut(index) or "") ~= "" then
```

Comment:
Split this onto a separate line to make it a bit clearer.

```suggestion
			local shortcut = C_ChatInfo.GetChannelShortcut(index);
			if not shortcut or shortcut == "" then
```

Reply:
This change inverts the logic. The if checks for a non empty shortcut (not nil and not empty string).
Correct change is :
```lua
local shortcut = C_ChatInfo.GetChannelShortcut(index) or "";
if shortcut ~= "" then
```